salt/states/archive.py

```diff
@@ -1239,153 +1440,205 @@ def extracted(name,
-                                # to dump the decompressed data to stdout and
-                                # pipe it to tar for extraction.
-                                cmd = 'xz --decompress --stdout {0} | tar xvf -'
-                                results = __salt__['cmd.run_all'](
-                                    cmd.format(_cmd_quote(cached)),
-                                    cwd=name,
-                                    python_shell=True)
-                                if results['retcode'] != 0:
-                                    if created_destdir:
-                                        _cleanup_destdir(name)
-                                    ret['result'] = False
-                                    ret['changes'] = results
-                                    return ret
-                                if _is_bsdtar():
-                                    files = results['stderr']
-                                else:
-                                    files = results['stdout']
-                            else:
-                                # Failed to open tar archive and it is not
-                                # XZ-compressed, gracefully fail the state
-                                if created_destdir:
-                                    _cleanup_destdir(name)
-                                ret['result'] = False
-                                ret['comment'] = (
-                                    'Failed to read from tar archive using '
-                                    'Python\'s native tar file support. If '
-                                    'archive is compressed using something '
-                                    'other than gzip or bzip2, the '
-                                    '\'options\' argument may be required to '
-                                    'pass the correct options to the tar '
-                                    'command in order to extract the archive.'
-                                )
-                                return ret
-                        else:
-                            if created_destdir:
-                                _cleanup_destdir(name)
-                            ret['result'] = False
-                            ret['comment'] = (
-                                'Failed to read from tar archive. If it is '
-                                'XZ-compressed, install xz-utils to attempt '
-                                'extraction.'
-                            )
-                            return ret
-                else:
-                    if not salt.utils.path.which('tar'):
-                        ret['comment'] = (
-                            'tar command not available, it might not be '
-                            'installed on minion'
-                        )
-                        return ret
+        if skip_same_file:
+            tmp_filename = salt.utils.hashutils.md5_digest(os.path.basename(cached))
```

Comment:
We should never explicitly use md5, mainly because it is not enabled in all openssl libs so this will fail on many installs.